x-pack/plugins/maps/public/classes/layers/tiled_vector_layer/tiled_vector_layer.tsx

```diff
@@ -160,6 +160,11 @@ export class TiledVectorLayer extends VectorLayer {
       return false;
     }
 
+    if (!mbTileSource.tiles) {
```

Comment:
This is because the layer-type and the underlying representation in mb changes when toggling between scaling-types. It needs to check it the mb-source is compatible with the layer-type.